### DISCUSSION TITLE
The component of y is missing.

### DISCUSSION THREAD

**Discussion by ngaiheibrian** [lululxvi/deepxde#1181] [Q&A]
Hello Sir, I am new to deepxde and try to apply it on my project. However, my code was failed to run.

It showed:

   raise ValueError("The component of y is missing.")
ValueError: The component of y is missing.

I would be grateful if you can point out which part of my code is wrong.

Thanks

"""Backend supported: tensorflow.compat.v1, tensorflow, pytorch, paddle, jax"""
import deepxde as dde
import matplotlib.pyplot as plt
import numpy as np
from scipy import integrate
# Import tf if using backend tensorflow.compat.v1 or tensorflow
from deepxde.backend import tf
# Import torch if using backend pytorch
# import torch
# Import paddle if using backend paddle
# import paddle
# Import jax.numpy if using backend jax
# import jax.numpy as jnp

def ode_system(x,y):
    m=0.2591
    k=100
    damping_ratio=0.05
    c=damping_ratio
    time_applied=1
    ui=193
    u=0
    
    y1  =  y[:,0:1]
    y2  =  y[:,1:2] 
    y3  =  y[:,2:3] 
    y4  =  y[:,3:4]
    y5  =  y[:,4:5]
    
    dy1_t= dde.grad.jacobian(y,x,i=0, j=0)
    dy2_t= dde.grad.jacobian(y,x,i=1, j=0)
    dy3_t= dde.grad.jacobian(y,x,i=2, j=0)
    dy4_t= dde.grad.jacobian(y,x,i=3, j=0)
    dy5_t= dde.grad.jacobian(y,x,i=4, j=0)

    dy1_tt =dde.grad.hessian(y, x, i=0, j=0)
    dy2_tt =dde.grad.hessian(y, x, i=1, j=0)
    dy3_tt =dde.grad.hessian(y, x, i=2, j=0)
    dy4_tt =dde.grad.hessian(y, x, i=3, j=0)
    dy5_tt =dde.grad.hessian(y, x, i=4, j=0)

    if x[:,0:1]<=time_applied:
        u=ui*tf.sin(2*np.pi*x[:, 0:1])

    else:
        u=0
    

    

    

    return [
            m*dy1_tt+c*dy1_t+k*(2*y1-y2)+m*u,
            m*dy2_tt+c*dy2_t+k*(-y1+2*y2-y3)+m*u,
            m*dy3_tt+c*dy3_t+k*(-y2+2*y3-y4)+m*u,
            m*dy4_tt+c*dy4_t+k*(-y3+2*y4-y5)+m*u,
            m*dy5_tt+c*dy5_t+k*(-y4+y5)+m*u
            ]

def boundary(_, on_initial):
    return on_initial

def func(x):
    """
    y1 = sin(x)
    y2 = cos(x)
    """
    return np.hstack(x)

geom = dde.geometry.TimeDomain(0, 20)
ic1 = dde.icbc.IC(geom, lambda x: 0, boundary, component=0)
ic2 = dde.icbc.IC(geom, lambda x: 0, boundary, component=1)
ic3 = dde.icbc.IC(geom, lambda x: 0, boundary, component=2)
ic4 = dde.icbc.IC(geom, lambda x: 0, boundary, component=3)
ic5 = dde.icbc.IC(geom, lambda x: 0, boundary, component=4)

data = dde.data.PDE(geom, ode_system, [ic1, ic2, ic3, ic4, ic5], 35, 2, solution=func, num_test=100)

layer_size = [1] + [50] * 3 + [5]
activation = "tanh"
initializer = "Glorot uniform"
net = dde.nn.FNN(layer_size, activation, initializer)

model = dde.Model(data, net)
model.compile("adam", lr=0.001, metrics=["l2 relative error"])
losshistory, train_state = model.train(iterations=20000)

dde.saveplot(losshistory, train_state, issave=True, isplot=True)

**Comment by ghost** [comment]
You have to put your code inside markdown code block.
See <https://deepxde.readthedocs.io/en/latest/modules/deepxde.html#deepxde.gradients.hessian>
```python
dy1_tt = dde.grad.hessian(y, x, component=0, i=0, j=0)
dy2_tt = dde.grad.hessian(y, x, component=1, i=0, j=0)
dy3_tt = dde.grad.hessian(y, x, component=2, i=0, j=0)
dy4_tt = dde.grad.hessian(y, x, component=3, i=0, j=0)
dy5_tt = dde.grad.hessian(y, x, component=4, i=0, j=0)

u = torch.where(torch.less(x[:, 0:1], time_applied), ui * torch.sin(2 * np.pi * x[:, 0:1]), 0)
```

**Reply by ngaiheibrian** [reply]
Thanks a lot, I finally sorted the problem!!

**Comment by ngaiheibrian** [comment]
And guys, I have another question. As I am working on a project that I need to train a model to predict a bridge motion which contains over 80k node and it is impossible to write the program using the method ie dy1 dy2..... , is there any tips for me to build a model using deepxde?

**Reply by lululxvi** [reply]
https://github.com/lululxvi/deepxde/issues/865
